web/app/dash/submissions/[id]/page.tsx

```diff
@@ -16,10 +17,30 @@ export default function Page({ params }: { params: { id: string } }) {
       router.push("/dash");
     }
   });
-  return (
-    <h1>
-      {userInfo?.user?.algos?.get(params.id)?.lintFailureMessage ||
-        userInfo?.user?.algos?.get(params.id)?.lintSuccessMessage || ""}
-    </h1>
-  );
+
+  const formatNewLines = (str: string) => {
+    const LINES = str.split("\n");
+    return LINES.map((line: string, index: number) => (
+      <React.Fragment key={`line_${index}`}>
+        <p>{line}</p>
+        {index < LINES.length - 1 && <br />}
+      </React.Fragment>
+    ));
+  }
+  
+
+  const algoDetails = userInfo?.user?.algos?.get(params.id);
+  const lintFailureMessage = algoDetails?.lintFailureMessage;
+  const lintSuccessMessage = algoDetails?.lintSuccessMessage;
+  const stringToRender = lintFailureMessage || lintSuccessMessage || "";
+
+  if(stringToRender === "") {
+    return <></>
+  } else {
+    return (
+      <>
+        {formatNewLines(stringToRender)}
+      </>
+    );
```

Comment:
i think it may be a good idea to return a `div` so you can apply styling

current the text does not have any margin/padding from the box its in iirc